### ISSUE TITLE
Add styles for form validation

### ISSUE THREAD

**Issue by heyallan** [heyallan/html.css#5]
**Status**

- Probably out of scope
- Further research is needed

**Comment by heyallan** [comment]
**Overview**

- Validated inputs are evaluated on load, before user interaction
- `:required` inputs are `:invalid` by default
- `:optional` inputs are `:valid` by default
- A combination of CSS selectors is needed for applying styles only after user interaction
- Available selectors:
  - `:valid` and `:invalid` limited control
  - `:focus` limited control
  - `:placeholder-shown` not enough support
  - "empty-value" selector is unavailable (-_-!)
    - Not to be confused with `:empty` which means "empty-node"

**CSS examples**

- Basic CSS: [Jim Nielsen tutorial demo](https://cdn.tutsplus.com/webdesign/uploads/legacy/tuts/214_html5_form_validation/demo/index.html)
- Advanced CSS: [Chris Coyier CodePen](http://codepen.io/chriscoyier/pen/JXgKjb)

**Possible approach**

Demo: http://jsbin.com/conota/edit?html,css,output

**Conclusion**

- CSS feedback is technically possible
- Not enough control for an optimal user experience
- Primary limitation is controlling/avoiding immediate feedback without user interaction

**Comment by heyallan** [comment]
Not enough selectors (yet) to create an optimal user experience
